test/unit/org/authInfoTest.ts

```diff
@@ -1409,6 +1411,61 @@ describe('AuthInfo', () => {
       expect(await AuthInfo.hasAuthentications()).to.be.true;
     });
   });
+  describe('getNamespacePrefix', () => {
+    it('should get the namespace associated to the org', async () => {
+      $$.setConfigStubContents('AuthInfoConfig', { contents: await testOrg.getConfig() });
+
+      stubMethod($$.SANDBOX, Transport.prototype, 'httpRequest')
+        .withArgs(
+          match({
+            url: `${testOrg.instanceUrl}//services/data/v51.0/query?q=Select%20Namespaceprefix%20FROM%20Organization`,
+          })
+        )
+        .resolves({
+          statusCode: 200,
+          body: JSON.stringify({
+            records: [
+              {
+                NamespacePrefix: `acme_${testOrg.testId}`,
+              },
+            ],
+          }),
+        });
+      const authInfo = await AuthInfo.create({ username: testOrg.username });
+      // @ts-expect-error because private method
+      expect(await authInfo.getNamespacePrefix(testOrg.instanceUrl, testOrg.accessToken)).to.equal(
+        `acme_${testOrg.testId}`
+      );
+
+      expect(authInfo.getFields().namespacePrefix).to.equal(`acme_${testOrg.testId}`);
+    });
+    it('should not set namespace prop if org doesn not have one', async () => {
+      const orgConfig = await testOrg.getConfig();
+      orgConfig.namespacePrefix = undefined;
+      $$.setConfigStubContents('AuthInfoConfig', { contents: orgConfig });
+
+      stubMethod($$.SANDBOX, Transport.prototype, 'httpRequest')
+        .withArgs(
+          match({
+            url: `${testOrg.instanceUrl}//services/data/v51.0/query?q=Select%20Namespaceprefix%20FROM%20Organization`,
+          })
+        )
+        .resolves({
+          statusCode: 200,
+          body: JSON.stringify({
+            records: [
+              {
+                NamespacePrefix: null,
+              },
+            ],
+          }),
+        });
+      const authInfo = await AuthInfo.create({ username: testOrg.username });
+      // @ts-expect-error because private method
+      expect(await authInfo.getNamespacePrefix(testOrg.instanceUrl, testOrg.accessToken)).to.be.undefined;
+      expect(authInfo.getFields().namespacePrefix).to.be.undefined;
```

Comment:
this test verifies that `getNamespacePrefix` returns undefined instead of an empty string to ensure we never end up with `namespacePrefix: ""` in an auth file.